### PR TITLE
Add missing worker settings to shared configuration

### PR DESCRIPTION
**Pull request by dklimpel**
Related to #13643 

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Dirk Klimpel 5740567+dklimpel@users.noreply.github.com

### CODE REVIEW

File: docs/usage/configuration/config_documentation.md [hunk 1]
```diff
@@ -3841,6 +3841,39 @@ Example configuration:
...
+The [worker](../../workers.md#synapseappmedia_repository) that is used to run
+background tasks for media repository. If running multiple media repositories
+you must configure a single instance to run the background tasks. If not provided
+this defaults to the main process or your single `media_repository` worker.
+
+Example configuration:
+```yaml
+media_instance_running_background_jobs: worker1
+```
+---
```
Comment (DMRobertson) [inline docs/usage/configuration/config_documentation.md:3876]: I think some of these are fairly recent. Can you say which Synapse release these were added in? The usual format is 

`_Added in Synapse X.YZ._`.

Looks like the relevant PRs are #12654, #12452, #7706.

File: docs/workers.md [hunk 2]
```diff
@@ -502,11 +504,32 @@ after setting this option in the shared configuration!
 This style of configuration supersedes the legacy `synapse.app.appservice`
 worker application type.
 
+#### Push Notifications
+
+You can designate one generic worker to sending push notifications to
+[sygnal](https://github.com/matrix-org/sygnal) and email.
```
Comment (DMRobertson) [inline docs/workers.md:510]: Can we mention push gateways here? e.g.

```
sending push notifications to a [push gateway](https://spec.matrix.org/v1.5/push-gateway-api/) such as [sygnal](...
```

File: docs/workers.md [hunk 3]
```diff
@@ -502,11 +504,32 @@ after setting this option in the shared configuration!
...
+This will stop the main synapse sending push notifications.
```
Comment (DMRobertson) [inline docs/workers.md:511]: "main process" rather than "main synapse" here, I think?

File: docs/workers.md [hunk 4]
```diff
@@ -502,11 +504,32 @@ after setting this option in the shared configuration!
 This style of configuration supersedes the legacy `synapse.app.appservice`
 worker application type.
 
+#### Push Notifications
+
+You can designate one generic worker to sending push notifications to
```
Comment (DMRobertson) [inline docs/workers.md:509]: is the word "one" correct here? the text below describing `pusher_instances` suggests that we can have multiple pusher workers.

File: docs/workers.md [hunk 5]
```diff
@@ -502,11 +504,32 @@ after setting this option in the shared configuration!
 This style of configuration supersedes the legacy `synapse.app.appservice`
 worker application type.
 
+#### Push Notifications
```
Comment (DMRobertson) [inline docs/workers.md:507]: Not sure if this was meant to be in this PR, but for what it's worth:

File: docs/workers.md [hunk 6]
```diff
@@ -502,11 +504,32 @@ after setting this option in the shared configuration!
...
+
+You can designate one generic worker to sending push notifications to
+[sygnal](https://github.com/matrix-org/sygnal) and email.
+This will stop the main synapse sending push notifications.
+Doesn't handle any REST endpoints itself, but you should specify its name in the
+[shared configuration](usage/configuration/config_documentation.md#pusher_instances)
+as follows:
```
Comment (DMRobertson) [inline docs/workers.md:514]: Not sure what you're driving at with the REST endpoints here. Is the point that you don't need to adjust reverse proxy config when configuring pusher workers? If so, how about something like

```
The workers responsible for sending push notifications can be defined using the `pusher_instances` option. For example:

...

Workers don't need to accept incoming HTTP requests to send push notifications, so no additional reverse proxy configuration is required for pusher workers.
```

**Review by DMRobertson — CHANGES_REQUESTED** [review]
Many thanks for this. I've left some suggestions:

File: docs/workers.md (hunk 2 above)
Comment (dklimpel) [reply docs/workers.md]: Done

File: docs/workers.md (hunk 3 above)
Comment (dklimpel) [reply docs/workers.md]: Done

File: docs/workers.md (hunk 6 above)
Comment (dklimpel) [reply docs/workers.md]: The information that a worker does not accept HTTP requests is available for different workers, therefore also here.

File: docs/workers.md [hunk 7]
```diff
@@ -502,11 +504,38 @@ after setting this option in the shared configuration!
...
+pusher_instances:
+  - pusher_worker1
+  - pusher_worker2
+```
+
+Multiple workers can be added to this map, in which case the work is balanced
+across them. Ensure the main process and all pusher workers are restarted after changing
+this option.
+
+This workers don't need to accept incoming HTTP requests to send push notifications,
```
Comment (DMRobertson) [inline docs/workers.md:529]:
```suggestion
These workers don't need to accept incoming HTTP requests to send push notifications,
```

**Review by DMRobertson — APPROVED** [review]
Looks great, thanks Dirk!

**Pull request merged**